bigtable/client/instance_admin_test.cc

```diff
@@ -488,8 +488,278 @@ TEST_F(InstanceAdminTest, CreateInstancePollReturnsFailure) {
+}
+
+/// @test Verify that `bigtable::InstanceAdmin::UpdateInstance` works.
+TEST_F(InstanceAdminTest, UpdateInstanceImmediatelyReady) {
+  using ::testing::_;
+  using ::testing::Invoke;
+
+  bigtable::InstanceAdmin tested(client_);
+
+  std::string expected_text = R"(
+name: 'projects/my-project/instances/test-instance'
+display_name: 'foo bar'
+state: READY
+type: PRODUCTION
+)";
+  btproto::Instance expected;
+  ASSERT_TRUE(
+      google::protobuf::TextFormat::ParseFromString(expected_text, &expected));
+  std::string update_mask_text = R"(
+paths: 'display_name'
+)";
+  google::protobuf::FieldMask update_mask;
+  ASSERT_TRUE(google::protobuf::TextFormat::ParseFromString(update_mask_text,
+                                                            &update_mask));
+
+  EXPECT_CALL(*client_, UpdateInstance(_, _, _))
+      .WillOnce(Invoke(
+          [&expected](grpc::ClientContext*,
+                      btproto::PartialUpdateInstanceRequest const& request,
+                      google::longrunning::Operation* response) {
+            auto const instance_name =
+                "projects/my-project/instances/test-instance";
+            EXPECT_EQ(instance_name, request.instance().name());
+            response->set_done(true);
+            response->set_name("operation-name");
+            auto any = bigtable::internal::make_unique<google::protobuf::Any>();
+            any->PackFrom(expected);
+            response->set_allocated_response(any.release());
+            return grpc::Status::OK;
+          }));
+
+  EXPECT_CALL(*client_, GetOperation(_, _, _)).Times(0);
+
+  auto future = tested.UpdateInstance(&expected, &update_mask);
+  auto actual = future.get();
+
+  std::string delta;
+  google::protobuf::util::MessageDifferencer differencer;
+  differencer.ReportDifferencesToString(&delta);
+  EXPECT_TRUE(differencer.Compare(expected, actual)) << delta;
+}
+
+/// @test Failures while polling in `bigtable::InstanceAdmin::UpdateInstance`.
+TEST_F(InstanceAdminTest, UpdateInstancePollRecoverableFailures) {
+  using ::testing::_;
+  using ::testing::Invoke;
+
+  bigtable::InstanceAdmin tested(client_);
+  EXPECT_CALL(*client_, UpdateInstance(_, _, _))
+      .WillOnce(Invoke([](grpc::ClientContext*,
+                          btproto::PartialUpdateInstanceRequest const& request,
+                          google::longrunning::Operation* response) {
+        auto const instance_name =
+            "projects/my-project/instances/test-instance";
+        EXPECT_EQ(instance_name, request.instance().name());
+        return grpc::Status::OK;
+      }));
+
+  std::string expected_text = R"(
+name: 'projects/my-project/instances/test-instance'
+display_name: 'foo bar'
+state: READY
+type: PRODUCTION
+)";
+  btproto::Instance expected;
+  ASSERT_TRUE(
+      google::protobuf::TextFormat::ParseFromString(expected_text, &expected));
+
+  std::string update_mask_text = R"(
+paths: 'display_name'
```

Comment:
See other comment about indentation of strings.

Reply:
fixed